onnxruntime/core/framework/session_state.cc

```diff
@@ -933,11 +933,12 @@ Status SessionState::SaveToOrtFormat(flatbuffers::FlatBufferBuilder& builder,
 
 #endif  // !defined(ORT_MINIMAL_BUILD)
 
-Status SessionState::CreateSubgraphSessionState() {
+Status SessionState::CreateSubgraphSessionState(bool allow_unassigned_node_ep) {
   for (auto& node : graph_.Nodes()) {
     for (auto& entry : node.GetAttributeNameToMutableSubgraphMap()) {
       const auto& ep = node.GetExecutionProviderType();
-      if (ep != kCpuExecutionProvider && ep != kCudaExecutionProvider) {
+      ORT_ENFORCE(!ep.empty() || allow_unassigned_node_ep, "Node execution provider type must not be empty.");
+      if (!ep.empty() && ep != kCpuExecutionProvider && ep != kCudaExecutionProvider) {
```

Comment:
Do we need this ORT_ENFORCE as well as the logic in VerifyEachNodeIsAssignedToAnEpImpl? Prefer one place doing validation if possible.